### PR TITLE
RBD namespace in default pool tests

### PR DESCRIPTION
**Pull request by pavankumarag**
This PR is to automate namespace test in default pool and required cli and workflow files to support the test. The tests covered as part of this PR are (1) creating a namespace in default pool.

Logs are attached here.

To extract and check logs:
(1) mkdir test 
(2) cd test
(3) cp <namespace_log.tar.gz> .
(4) tar -xvzf namespace_log.tar.gz

[namespace_log.tar.gz](https://github.com/red-hat-storage/cephci/files/14214234/namespace_log.tar.gz)

image

### CODE REVIEW

File: suites/quincy/rbd/tier-2_rbd_namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      destroy-clster: false
+      module: test_cephadm.py
+      name: deploy cluster
+
+  - test:
+      abort-on-fail: true
+      config:
+        command: add
+        id: client.1
+        node: node10
```
Comment (sunilangadi2) [inline suites/quincy/rbd/tier-2_rbd_namespace.yaml:53]: which conf you used to test `node10` ?

File: suites/quincy/rbd/tier-2_rbd_namespace.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+# Test to be executed
+  - test:
+      desc: Run namespace creation in default pool
+      module: test_rbd_namespace_default_pool.py
+      name: RBD namespace creation in default pool
+      config:
+        rep-pool-only: True
+        rep_pool_config:
+          rbd: {}
```
Comment (sunilangadi2) [inline suites/quincy/rbd/tier-2_rbd_namespace.yaml:72]: why empty dict, please pass parameters appropriately?

File: tests/rbd/test_rbd_namespace_default_pool.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+from ceph.rbd.workflows.cleanup import cleanup
+from ceph.rbd.workflows.namespace import create_namespace_and_verify
+from ceph.utils import get_node_by_id
+from cli.rbd.rbd import Rbd
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_namespace_creation_default_pool(client, **kw):
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_namespace_default_pool.py:11]: please add method doc string with args

File: suites/quincy/rbd/tier-2_rbd_namespace.yaml (hunk 2 above)
Comment (sunilangadi2) [reply suites/quincy/rbd/tier-2_rbd_namespace.yaml:72]: also add poalrion id here

**Review by sunilangadi2 — CHANGES_REQUESTED** [review]
Please fix tox failures.

File: suites/quincy/rbd/tier-2_rbd_namespace.yaml (hunk 1 above)
Comment (pavankumarag) [reply suites/quincy/rbd/tier-2_rbd_namespace.yaml]: fixed

File: suites/quincy/rbd/tier-2_rbd_namespace.yaml (hunk 2 above)
Comment (pavankumarag) [reply suites/quincy/rbd/tier-2_rbd_namespace.yaml:85]: this is bcz the current non support to have image non creation from yaml structure

File: tests/rbd/test_rbd_namespace_default_pool.py (hunk 3 above)
Comment (pavankumarag) [reply tests/rbd/test_rbd_namespace_default_pool.py:11]: added

**Comment by pavankumarag** [comment]
Run summary
image

File: suites/quincy/rbd/tier-2_rbd_namespace.yaml [hunk 4]
```diff
@@ -0,0 +1,85 @@
+# Tier2: Added RBD namespace creation in default pool testing
```
Comment (sunilangadi2) [inline suites/quincy/rbd/tier-2_rbd_namespace.yaml:1]: please make it as `Tier2: Test suite to cover RBD namespace related tests.`
while adding new test related to namespace use same suite.

File: suites/quincy/rbd/tier-2_rbd_namespace.yaml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+# Tier-level: 2
+# Test-Suite: tier-2_rbd_namespace.yaml
+#
+# Ceph RBD component.
```
Comment (sunilangadi2) [inline suites/quincy/rbd/tier-2_rbd_namespace.yaml:5]: please remove this line it's redundant.

File: suites/reef/rbd/tier-2_rbd_namespace.yaml [hunk 6]
```diff
@@ -0,0 +1,85 @@
+# Tier2: Added RBD namespace creation in default pool testing
```
Comment (sunilangadi2) [inline suites/reef/rbd/tier-2_rbd_namespace.yaml:1]: same for this suite as well.

**Review by sunilangadi2 — COMMENTED** [review]
Other changes looks good.

**Review by sunilkumarn417 — DISMISSED** [review]
LGTM

File: suites/quincy/rbd/tier-2_rbd_namespace.yaml (hunk 5 above)
Comment (pavankumarag) [reply suites/quincy/rbd/tier-2_rbd_namespace.yaml:5]: :nit: @sunilangadi2 this is just a comment, should be fine, let me know if you disagree

File: suites/quincy/rbd/tier-2_rbd_namespace.yaml (hunk 4 above)
Comment (pavankumarag) [reply suites/quincy/rbd/tier-2_rbd_namespace.yaml]: done

**Review by sunilkumarn417 — APPROVED** [review]

**Review by sunilangadi2 — APPROVED** [review]

**Pull request merged**